subworkflows/local/generate_downstream_samplesheet.nf

```diff
@@ -0,0 +1,117 @@
+
+                            return [ id, group, single_end, fastq ]
+                    }.join ( ch_combined_assemblies_remap )
+                    .map {
+                        id, group, single_end, fastq, fasta ->
+                            def reads   = fastq instanceof List ? fastq.flatten() : [ fastq ]
+                            def meta    = [:]
+
+                            meta.id         = id
+                            meta.group      = group
+                            meta.single_end = single_end
+                            meta.fastq_1    = reads[0]
+                            meta.fastq_2    = !meta.single_end ? reads[1] : ''
+                            meta.fasta      = fasta ? fasta : ''
+
+                            return meta
+                    }
+                    .set { ch_mag_metadata }
+            } else {
+                // if coassembly was used, join FastQ and FastA by group
+                ch_combined_assemblies_remap = ch_combined_assemblies
+                    .map {
+                        meta, fasta ->
+                            def group = meta.id.split('group-')
+
+                                return [ group[1], fasta ]
+                    }
+                short_reads
+                    .map {
+                        meta, fastq ->
+                            def id          = meta.id
+                            def group       = meta.group
+                            def single_end  = meta.single_end
+
+                                return [ group, id, single_end, fastq ]
+                    }
+                    .combine ( ch_combined_assemblies_remap, by:0 )
+                    .map {
+                        group, id, single_end, fastq, fasta ->
+                            def reads   = fastq instanceof List ? fastq.flatten() : [ fastq ]
+                            def meta    = [:]
+
+                            meta.id         = id
+                            meta.group      = group
+                            meta.single_end = single_end
+                            meta.fastq_1    = reads[0]
+                            meta.fastq_2    = !meta.single_end ? reads[1] : ''
+                            meta.fasta      = fasta ? fasta : ''
+
+                            return meta
+                    }
+                    .set { ch_mag_metadata }
+            }
+
+            // Create samplesheet for each sample using meta information
+            ch_mag_id_samplesheets = ch_mag_metadata.collectFile() { meta ->
+                [ "${meta.id}_phageannotator_samplesheet.csv", "sample,group,fastq_1,fastq_2,fasta" + '\n' + "${meta.id},${meta.group},${meta.fastq_1},${meta.fastq_2},${meta.fasta}" + '\n' ]
+            }
+
+            // Merge samplesheet across all samples for the pipeline
+            ch_mag_id_samplesheets.collectFile(name: "phageannotator_samplesheet.csv", keepHeader:true, skip:1, storeDir:"${params.outdir}/downstream_samplesheets/")
```

Comment:
@jfy133 Just to clarify, are you thinking I should add the `storeDir:"${params.outdir}/downstream_samplsheets/"` to the first `collectFile()`?

Reply:
No, more don't bind just `fasta` to the string on the line above.

`params.outdir + <path I don't remember> + fasta.name` or something

Reply:
Or even just validate that if you turn on samplesheet creation, then you must also save the fastqs.

But yes, one way or another - running from explicitly saved fsstqs in outdir is preferable.

If you have `cleanup = true` in a config, something many do but the user never sees, they won't have any fastq files in the work dir to point to in the sample sheet 😬. This is harder to control for rather than directing the user to make sure to save the fsstqs